system/jhtree/ctfile.hpp

```diff
@@ -96,7 +96,8 @@ struct __declspec(novtable) jhtree_decl KeyHdr
     __int64 blobHead; /* fpos of first blob node f0x */
     __int64 metadataHead; /* fpos of first metadata node f8x */
     __int64 bloomHead; /* fpos of bloom table data, if present 100x */
-    __uint64 partitionFieldMask; /* Bitmap indicating partition keyed fields */
+    __uint64 partitionFieldMask; /* Bitmap indicating partition keyed fields 108x */
+    __int64 firstLeaf; /* fpos of first leaf node 110x */
```

Comment:
is there any compatibilty issues with a new key with these extra 8 bytes in header,
and old version reading them, due to the way getSize() is implemented, i.e. 
```
    inline static size32_t getSize() { return sizeof(KeyHdr); }
```

Reply:
A old system reading a new key will not even know about the existence of the new field, and will process the index as it always was.  A new system reading an old key will see the value initialised to -1, since all unused bytes after the header are set to 0xff.